src/guide/scaling-up/testing.md

```diff
@@ -1,32 +1,75 @@
 # Testing
 
-## Overview
+## Why Test?
+
+Automated tests help you and your team build complex Vue applications quickly and confidently by preventing regressions and encouraging you to break apart your application into testable functions, modules, classes, and components. As with any application, your new Vue app can break in many ways, and it's important that you can catch these issues and fix them before releasing.
 
-When it comes to building reliable applications, automated tests can play a critical role in an individual or team's ability to build new features, refactor code, and fix bugs with confidence.
+In this guide, we'll cover basic terminology and provide our recommendations on which tools to choose for your Vue 3 application.
 
-In a Vue application, there are different types of concerns that can be covered by automated tests:
+There is one Vue-specific section covering composables. See [Testing Composables](#testing-composables) below for more details.
 
-- **Logic**: verify non-UI rendering logic is implemented correctly. This typically involves application-wide business logic (e.g. state management stores) and utility functions.
+:::tip Start testing early!
+We recommend you begin writing tests as soon as you can. The longer you wait to add tests to your application, the more dependencies your application will have, and the harder it will be to start.
+:::
 
-- **Visual**: given certain input props / state, verify that an isolated component or component tree is rendering the correct visual output.
+## Testing Types
 
-- **Interaction**: simulate user behavior, such as a click or input, and verify that an isolated component or component tree renders correct, updated output.
+When designing your Vue application's testing strategy, you should leverage the following testing types:
 
-- **User Flow**: given a sequence of user interactions that are necessary to complete an actual user task, check whether the application as a whole is working as expected.
+- **Unit**: Checks that inputs to a given function, class, or composable are producing the expected output or side effects.
+- **Component**: Checks that your component mounts, renders, can be interacted with, and behaves as expected. These tests import more code than unit tests, are more complex, and require more time to execute.
+- **End-to-end**: Checks features that span multiple pages and make real network requests against your production-built Vue application. These tests often involve standing up a database or other backend.
```

Comment:
**suggestion**: I believe we're missing an `Integration testing` here

Reply:
Maybe we do that as a follow-up. I really do want to include the Testing Pyramid to explain how speed/complexity are all relative, but I don't want to rework the sections _again_ right now.